cmds/webboot/webboot.go

```diff
@@ -66,6 +78,36 @@ func (d *DownloadByLink) Exec() error {
 	return nil
 }
 
+// Exec let user to choose a group of the cached iso
+func (i *InstallCachedISO) Exec() error {
+	isos := i.cachedISO
+	groupedIso := make(map[string]([]*CachedISO))
```

Comment:
I wonder if having the 'group' be a field of a struct and having to iterate through all ISOs to organize by group every time this menu is presented is a little inefficient. Is there a way we can do this in levels instead? So I'm imaging a directory structure for the cached ISOs that looks something like this

```
TinyCore/
   TinyCorePure64.iso
Ubuntu/
   Ubuntu-20.04.iso
   Ubuntu-18.04.iso
ArchLinux/
```
So depending on how you've worked out the directory structure expected on the USB device, would it make sense to have the menu display one level of directories at a time? Then once one is selected, the Exec for that can find subdirectories / isos in the given directory and show the next menu? That way you're not having to do all this classification every time. I think this might also simplify your types.

@rminnich any thoughts on this?

Reply:
I think that's a good idea! thank you !